sklearn/neighbors/tests/test_ball_tree.py

```diff
@@ -85,3 +85,13 @@ def test_array_object_type():
     X = np.array([(1, 2, 3), (2, 5), (5, 5, 1, 2)], dtype=object)
     with pytest.raises(ValueError, match="setting an array element with a sequence"):
         BallTree(X)
+
+
+def test_bad_pyfunc_metric():
+    def wrong_distance(x):
+        return "1"
```

Comment:
```suggestion
        return 1.0
```

Reply:
We could also add another test for the case  where `wrong_distance` accept 2 vectors and returns a string and check that we raise a meaningful error message in that case but this outside of the original scope of this PR.

Reply:
Yes, this test was removed by mistake in https://github.com/scikit-learn/scikit-learn/pull/21177/files#diff-7d6d6311a62ec641133e6fde00dee4cdd4dc8c56bb37c49b398ccfec8383eaa7L233-L240 and is being reintroduced here.

Reply:
Resolving as cb32715 fixed the problem.